src/native/managed/cdacreader/src/Contracts/Metadata.MethodTableFlags.cs

```diff
@@ -0,0 +1,190 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using UntrustedMethodTable = Microsoft.Diagnostics.DataContractReader.Contracts.UntrustedMethodTable_1;
+using MethodTable = Microsoft.Diagnostics.DataContractReader.Contracts.MethodTable_1;
+using UntrustedEEClass = Microsoft.Diagnostics.DataContractReader.Contracts.UntrustedEEClass_1;
+using EEClass = Microsoft.Diagnostics.DataContractReader.Contracts.EEClass_1;
+
+namespace Microsoft.Diagnostics.DataContractReader.Contracts;
+
+internal partial struct Metadata_1
+{
+    [Flags]
+    internal enum WFLAGS_LOW : uint
+    {
+        // We are overloading the low 2 bytes of m_dwFlags to be a component size for Strings
+        // and Arrays and some set of flags which we can be assured are of a specified state
+        // for Strings / Arrays, currently these will be a bunch of generics flags which don't
+        // apply to Strings / Arrays.
+
+        UNUSED_ComponentSize_1 = 0x00000001,
+        // GC depends on this bit
+        HasCriticalFinalizer = 0x00000002, // finalizer must be run on Appdomain Unload
```

Comment:
Similar to my comment from other PR. Do all these bits need to be part of the diagnostic contract?

Reply:
I need to do a bit of refactoring.

There are two sides to the contract:
* the information that cdac consumes from the runtime.  I think probably we don't need most of the bits from the method table flags.  I'll remove the ones we don't need
* the public contract surface.  Right now `MethodTable_1` is leaky because it does actually expose the flags dwords completely.   I think I don't actually like having `MethodTable_1` having any public properties - all public interaction with it should be through a `Metadata`  contract subclass which should only expose the logical interpretation of the flags bits ("is string", "is generic instance", etc)